File: Examples/test-suite/java/director_rev_len_str_runme.java

import director_rev_len_str.*;

public class director_rev_len_str_runme {

  static {
    try {
      System.loadLibrary("director_rev_len_str");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {
    Callback callback = new DirectorReverseLengthStringCallback();
    Caller caller = new Caller(callback);
    int sum = caller.call();

    if (sum != 9*2*8)
      throw new RuntimeException("Unexpected sum: " + sum);
  }
}

class DirectorReverseLengthStringCallback extends Callback {
  public DirectorReverseLengthStringCallback() {
    super();
  }

  @Override
  public int run(String dataBufferAA)
  {
    int ret = 0;
    if (dataBufferAA != null) {
      for (int i = 0; i < dataBufferAA.length(); i++)
        ret += (int)dataBufferAA.charAt(i) * 2;
    }
    return ret;
  }
}

File: Examples/test-suite/octave/director_rev_len_str_runme.m
director_rev_len_str

function self=DirectorReverseLengthStringCallback()
  global director_rev_len_str;
  self=subclass(director_rev_len_str.Callback());
  self.run=@DirectorReverseLengthStringCallback_run;
end
function ret=DirectorReverseLengthStringCallback_run(self, dataBufferAA)
  ret = 0;
  for i = 1:length(dataBufferAA)
    ret = ret + double(dataBufferAA(i)) * 2;
  end
end

callback = DirectorReverseLengthStringCallback();
caller = director_rev_len_str.Caller(callback);
sum = caller.call();

if (sum != 9*2*8)
    error(sum);
end
